lib/fillalgorithms.cpp

```diff
@@ -126,6 +126,29 @@ QVector<QPoint> imagePixelFloodFill(const QImage *image, const QPoint &startPos,
     return filledPositionsVector;
 }
 
+struct ImagePixelHash {
+    explicit ImagePixelHash(int hashSize, int imageWidth) : mImageWidth{imageWidth}, mData(hashSize, 0) {}
+
+    int getImageIndex(const QPoint& point) {
```

Comment:
Would it really hurt performance even if they weren't inlined? In general I think it's just easier to always pass non-primitive types by const reference, rather than trying to remember the memory usage of each type. :D

Reply:
Well, it's certainly easier to always do the same thing, but it more sense to me to be roughly aware of the cost of copying vs. the cost of referencing. And in case copying is trivial and no more data is copied than the size of a reference, then I would always prefer a pass by value.

But, I did not measure it so I can't really answer your question...